sdcm/wait.py

```diff
@@ -70,11 +74,17 @@ def retry_logger(retry_state):
 
     except Exception as ex:  # pylint: disable=broad-except
```

Comment:
what about this?

```
    except Exception as ex:  # pylint: disable=broad-except
        if stop_event and stop_event.is_set():
            err = f": {text or func.__name__}: stopped by Event"
            raise WaitForStopEventtError(err) from ex
        else:
            err = f"Wait for: {text or func.__name__}: timeout - {timeout} seconds - expired"
            LOGGER.error(err)
            if hasattr(ex, 'last_attempt') and ex.last_attempt.exception() is not None:  # pylint: disable=no-member
                LOGGER.error("last error: %r", ex.last_attempt.exception())  # pylint: disable=no-member
            else:
                LOGGER.error("last error: %r", ex)
                res = ex.last_attempt._result if hasattr(  # pylint: disable=protected-access,no-member
                    ex, 'last_attempt') else None

            if throw_exc:
                if hasattr(ex, 'last_attempt') and not ex.last_attempt._result:  # pylint: disable=protected-access,no-member
                    raise WaitForTimeoutError(err) from ex
```

and now the function will have several scenarios
not changed:
return result - when func returns result
WaitForTimeoutError - when func returns an error, the event is not set(or none), timeout is exceeded
new:
WaitForStopEventtError - when func returns an error but the event is set, and SCT code will expect this error  to do not "raise any exception or critical event,"
It will be like just one more condition of exit without touching existing logic 

also, as I understood, exit by the event will be some kind of positive exit. so I think LOGGER.error is not needed in this case